scripts/parse_nagim.py

```diff
@@ -0,0 +1,467 @@
+        return 'uploaded'
+
+
+def _get_bucket_ls(
+    ext: Union[str, List[str]],
+    source_bucket,
+    tmp_dir,
+    overwrite,
+) -> List[str]:
+    label = ext.replace('.', '-')
+    output_path = join(tmp_dir, f'gs-ls{label}.txt')
+    if overwrite or not exists(output_path):
+        _call(f'test -e {output_path} && rm {output_path}')
+        _call(f'touch {output_path}')
+        if isinstance(ext, str):
+            ext = [ext]
+        for e in ext:
+            _call(f'gsutil ls "{source_bucket}/*{e}" >> {output_path}')
+    with open(output_path) as f:
+        return [line.strip() for line in f.readlines() if line.strip()]
+
+
+def _find_upload_files(
+    upload_gvcf_bucket,
+    upload_cram_bucket,
+    tmp_dir,
+    overwrite=False,
+) -> List[Sample]:
+    gvcf_paths = _get_bucket_ls(
+        ext='.vcf.gz',
+        source_bucket=upload_gvcf_bucket,
+        tmp_dir=tmp_dir,
+        overwrite=overwrite,
+    )
+    tbi_paths = _get_bucket_ls(
+        ext='.vcf.gz.tbi',
+        source_bucket=upload_gvcf_bucket,
+        tmp_dir=tmp_dir,
+        overwrite=overwrite,
+    )
+    cram_paths = _get_bucket_ls(
+        ext='.cram',
+        source_bucket=upload_cram_bucket,
+        tmp_dir=tmp_dir,
+        overwrite=overwrite,
+    )
+    crai_paths = _get_bucket_ls(
+        ext='.cram.crai',
+        source_bucket=upload_cram_bucket,
+        tmp_dir=tmp_dir,
+        overwrite=overwrite,
+    )
+
+    file_by_type_by_sid = {
+        'gvcf': dict(),
+        'tbi': dict(),
+        'cram': dict(),
+        'crai': dict(),
+    }
+
+    for fname in gvcf_paths:
+        sid = os.path.basename(fname).replace('.hard-filtered.g.vcf.gz', '')
+        file_by_type_by_sid['gvcf'][sid] = fname
+
+    for fname in tbi_paths:
+        sid = os.path.basename(fname).replace('.hard-filtered.g.vcf.gz.tbi', '')
+        if sid not in file_by_type_by_sid['gvcf']:
+            logger.info(f'Found TBI without GVCF: {fname}')
+        else:
+            file_by_type_by_sid['tbi'][sid] = fname
+
+    for fname in cram_paths:
+        sid = os.path.basename(fname).replace('.cram', '')
+        file_by_type_by_sid['cram'][sid] = fname
+
+    for fname in crai_paths:
+        sid = os.path.basename(fname).replace('.cram.crai', '')
+        if sid not in file_by_type_by_sid['cram']:
+            logger.info(f'Found CRAI without CRAM: {fname}')
+        else:
+            file_by_type_by_sid['crai'][sid] = fname
+
+    samples = []
+    for sid in file_by_type_by_sid['gvcf']:
+        samples.append(
+            Sample(
+                nagim_id=sid,
+                gvcf=file_by_type_by_sid['gvcf'][sid],
+                cram=file_by_type_by_sid.get('cram', {}).get(sid),
+            )
+        )
+    return samples
+
+
+cli.add_command(transfer)
+cli.add_command(parse)
+
+if __name__ == '__main__':
+    # pylint: disable=no-value-for-parameter
+    cli(obj={})  # pylint: disable=unexpected-keyword-arg
```

Comment:
Isn't this generally:
```suggestion
    cli()  # pylint: disable=unexpected-keyword-arg
```

Reply:
Remaining artefact from sharing context between the commands